docs/user-guide/scaling.md

```diff
@@ -0,0 +1,101 @@
+
+In practice, this ideal scenario may not be achieved, for a number of reasons.
+Firstly, you need to make sure you're using a parallel executor not the default single-threaded executor.
+You also need enough parallelism to match the scale of your problem.
+Weak scaling requires more workers than output chunks, so for large problems it might be necessary to adjust the executor's configuration to not restrict the ``max_workers``.
+With fewer workers than chunks we would expect linear strong scaling, as every new worker added has nothing to wait for.
+Stragglers are tasks that take much longer than average, who disproportionately hold up the next step of the computation.
+To handle stragglers, you should consider turning on backups, as any failures that are restarted essentially become stragglers.
+Worker start-up time is another practical speed consideration, though it would delay computations of all scales equally.
+
+### Multi-step Calculation
+
+A multi-step calculation requires writing one or more intermediate arrays to persistent storage.
+One important example in Cubed is the [rechunk](Operations/rechunk) operation, which guarantees bounded memory usage by writing and reading from one intermediate persistent Zarr store.
+
+In multi-step calculations, the number of steps in the plan sets the minimum total execution time.
+Hence, reducing the number of steps in the plan can lead to significant performance improvements.
+Reductions can be carried out in fewer iterative steps if ``allowed_mem`` is larger.
+Cubed automatically fuses some steps to enhance performance, but others (especially rechunk) cannot be fused without requiring a shuffle, which can potentially violate memory constraints.
+
+```{note} In theory multiple blockwise operations can be fused together, enhancing the performance further. However this has not yet been implemented in Cubed.
+```
+
+In practical scenarios, stragglers can hold up the completion of each step separately, thereby cumulatively affecting the overall performance of the calculation.
+
+### Multi-pipeline Calculation
+
+A "pipeline" refers to an independent branch of the calculation.
+For example, if you have two separate arrays to compute simultaneously, full parallelism requires sufficient workers for both tasks.
+The same logic applies if you have two arrays feeding into a single array or vice versa.
+
+
+```{note} Currently Cubed will not necessarily execute independent pipelines in parallel on all executors.
+```
+
+
+## Other Performance Considerations
+
+### Different Executors
+
+Different executors come with their own set of advantages and disadvantages.
+For instance, some executors have much faster worker startup times, like Modal, while others may have different limits to the maximum workers.
+When using Dask as an executor, the scheduler might yield unintuitive results.
```

Comment:
Can you explain why Dask might give unintuitive results, or give an example?

Reply:
I'm not what I was thinking when I wrote this, I've removed it in favour of talking about there being differences between running on a cluster and running on serverless instead.